### DISCUSSION TITLE
Exporting Teleport users

### DISCUSSION THREAD

**Discussion by jnixon55** [gravitational/teleport#8255] [Q&A]
I was tasked with creating a new teleport server and was wondering if there is a way to export teleport users, we are currently using an old version of teleport 4.4 and want to move to a new os and new version of teleport

**Comment by programmerq** [comment]
You can indeed export existing teleport users. On one of your auth servers, run `tctl get users --with-secrets > userexport.yaml` and that will grab all your teleport users and their corresponding password hashes and totp values. You can then copy that file over to a different cluster and run `tctl create -f userexport.yaml`.

The "Migrating backends" section of the docs covers this and other teleport resources you may want to migrate as well: https://goteleport.com/docs/admin-guide/#migrating-backends

**Reply by jnixon55** [reply]
Thanks for the reply, looks like thats not working for us. We are using OTP (google auth) can we manually create a QR code after the user has been created or is that just created the first time the user is created. I find myself having to delete and re-create a user just to get a new QR code.
